### DISCUSSION TITLE
Understand timezone for importing data

### DISCUSSION THREAD

**Discussion by Dany1410** [kimai/kimai#3388] [Questions & Answers]
Hello everybody,

I was just about to import all my data from Kimai Version 1 to Kimai Version 2. 
During my import I got the following response:
![error daten import](https://user-images.githubusercontent.com/107126760/176149459-e9e15faf-89c0-416d-9c69-7ec182c60441.PNG)

I used the following command for importing the data:
bin/console kimai:import:v1 --global --timezone="timezone" --language="language" --country="DE" --currency="EUR" --prefix="neu_" --fix-email=example.com --fix-timesheet "mysql://kimai_user:blabla@127.0.0.1:3306/kimai_db?charset=utf8" "blabla"

Instead of putting "timezone" I also tried "Europe", but this didn't help.

How is the timezone taking effect on my existing records of my timesheet?
Does anybody know, how to get rid of this errors, so I can import all records and don't leave those aprox. 14.000 records behind?

Thank you in advance!

Best 
Daniel

**Comment by kevinpapst** [comment]
That is weird, can't imagine why only 14k of 60k would not work. 
A timezone is a string like `Europe/Berlin` `Europe/Vienna` `Europe/Paris`. 
It is important to set the correct one, otherwise your imported times will be shifted by the timezone offset (eg. 2 hours for middle Europe).

Assuming that no private person is able to ever create 60k records and you are running a company: I offer paid support.

**Reply by Dany1410** [reply]
Hi Kevin,

Thank you for your support.
First I would try some things in my Database. 
I'll get back to you.

**Comment by Dany1410** [comment]
I was able to fix it by setting all columns "visible" in tables projects, activities and customers to visible=1.
Only 15 rows where left behind which is not a gib deal.

**Reply by kevinpapst** [reply]
Hm, so you are saying some validator is kicking in and denying to create timesheets for hidden customers/projects? 
That sounds weird, I am importing data regularly and never saw such a phenomenon. But thanks for sharing your solution, I will keep an eye on it!

**Reply by Dany1410** [reply]
Yeah, that is most probably what I was intended to say :D
I'll probably test the data import a few more times. Maybe I find a solution to disable the validation of records during the import.

**Reply by kevinpapst** [reply]
After checking the code I'd say I simply never had a customer using the "disabled project" feature in Kimai 1.
All imported records will be validated like regular timesheets created with the UI and that is not allowed with hidden customers/projects:
https://github.com/kevinpapst/kimai2/blob/master/src/Command/KimaiImporterCommand.php#L1680

We should probably have an ignore-list in there, with a couple of validation issues that can be skipped.

**Reply by Dany1410** [reply]
That could make sense. But, if you have seen this for the first time is it worth it expanding the code to add this feature?
When I import my data and have to make some some manuel adjustments, thats fine by me. And those remaining 15 rows are not decisive - at least for us.
